wradlib/georef/raster.py

```diff
@@ -217,57 +191,125 @@ def read_gdal_values(dataset=None, nodata=None):
+        raster image with georeferencing (GeoTransform at least)
+    geo : bool
+        True to get geographical coordinates
+    window : bool
+        True to get the window containing the corners
 
-    Note
-    ----
-    The origin of the provided data and coordinates is UPPER LEFT.
+    Returns
+    -------
+    extent : :class:`numpy:numpy.ndarray`
+        corner coordinates [ul,ll,lr,ur] or
+        window extent [xmin, xmax, ymin, ymax]
     """
 
-    # align data
-    data = data.copy()
-    if data.ndim == 2:
-        data = data[np.newaxis, ...]
-    bands, rows, cols = data.shape
+    x_size = dataset.RasterXSize
+    y_size = dataset.RasterYSize
+    geotrans = dataset.GetGeoTransform()
+    xmin = geotrans[0]
+    ymax = geotrans[3]
+    xmax = geotrans[0] + geotrans[1] * x_size
+    ymin = geotrans[3] + geotrans[5] * y_size
 
-    # create In-Memory Raster with correct dtype
-    mem_drv = gdal.GetDriverByName('MEM')
-    gdal_type = gdal_array.NumericTypeCodeToGDALTypeCode(data.dtype)
-    dataset = mem_drv.Create('', cols, rows, bands, gdal_type)
+    extent = np.array([[xmin, ymax],
+                       [xmin, ymin],
+                       [xmax, ymin],
+                       [xmax, ymax]])
 
-    # initialize geotransform
-    x_ps, y_ps = coords[1, 1] - coords[0, 0]
-    geotran = [coords[0, 0, 0], x_ps, 0, coords[0, 0, 1], 0, y_ps]
-    dataset.SetGeoTransform(geotran)
+    if geo:
+        projection = read_gdal_projection(dataset)
+        fun = georef.projection.reproject
+        extent = fun(extent, projection_source=projection)
 
-    if projection:
-        dataset.SetProjection(projection.ExportToWkt())
+    if window:
+        x = extent[:, 0]
+        y = extent[:, 1]
+        extent = np.array([x.min(), x.max(), y.min(), y.max()])
 
-    # set np.nan to nodata
-    dataset.GetRasterBand(1).SetNoDataValue(nodata)
+    return(extent)
 
-    for i, band in enumerate(data, start=1):
-        dataset.GetRasterBand(i).WriteArray(band)
 
-    return dataset
+def get_raster_elevation(dataset, **kwargs):
+    """Return surface elevation corresponding to raster dataset
+       The resampling algorithm is chosen based on scale ratio
+
+    Parameters
+    ----------
+    dataset : gdal.Dataset
+        raster image with georeferencing (GeoTransform at least)
+    kwargs : keyword arguments
+        passed to wradlib.io.dem.get_strm()
+
+    Returns
+    -------
+    elevation : :class:`numpy:numpy.ndarray`
+        Array of shape (rows, cols, 2) containing elevation
+    """
+    extent = get_raster_extent(dataset)
+    src_ds = wradlib.io.dem.get_srtm(extent, **kwargs)
+
+    driver = gdal.GetDriverByName('MEM')
+    dst_ds = driver.CreateCopy('ds', dataset)
+
+    src_gt = src_ds.GetGeoTransform()
+    dst_gt = src_ds.GetGeoTransform()
+    src_scale = min(abs(src_gt[1]), abs(src_gt[5]))
+    dst_scale = min(abs(dst_gt[1]), abs(dst_gt[5]))
+    ratio = dst_scale/src_scale
+    resample = gdal.GRA_Bilinear
```

Comment:
Would be nice to make `resample` a kwarg which overrides the built-in behaviour.